File: backend/src/main/java/com/example/demo/bean/Interface.java
package com.example.demo.bean;

import java.util.List;

public class Interface extends Line{
	private int no;	//交互编号
	private String name;	//交互名称
	private String description;	//交互内容
	private String from;
	private String to;
	private List<Phenomenon> phenomenonList;	//现象列表
	private int x1;	//位置信息
	private int y1;
	private int x2;
	private int y2;
	public int getNo() {
		return no;
	}
	public void setNo(int no) {
		this.no = no;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public String getFrom() {
		return from;
	}
	public void setFrom(String from) {
		this.from = from;
	}
	public String getTo() {
		return to;
	}
	public void setTo(String to) {
		this.to = to;
	}
	public List<Phenomenon> getPhenomenonList() {
		return phenomenonList;
	}
	public void setPhenomenonList(List<Phenomenon> phenomenonList) {
		this.phenomenonList = phenomenonList;
	}
	public int getX1() {
		return x1;
	}
	public void setX1(int x1) {
		this.x1 = x1;
	}
	public int getY1() {
		return y1;
	}
	public void setY1(int y1) {
		this.y1 = y1;
	}
	public int getX2() {
		return x2;
	}
	public void setX2(int x2) {
		this.x2 = x2;
	}
	public int getY2() {
		return y2;
	}
	public void setY2(int y2) {
		this.y2 = y2;
	}
	public void refreshPhenomenonList(String oldShortName,String newShortName){
		for(Phenomenon phenomenon:phenomenonList){
			if(phenomenon.getFrom().equals(oldShortName)) phenomenon.setFrom(newShortName);
    		else if(phenomenon.getTo().equals(oldShortName)) phenomenon.setTo(newShortName);
		}
	}

}

File: frontend/src/app/entity/Interface.ts
import { Phenomenon } from './Phenomenon';
import { Line } from './Line'
export class Interface extends Line{
	no: number;	
	name: string;	
	description: string;	
	from: string;
	to: string;
	phenomenonList: Phenomenon[];	
	x1: number;	
	y1: number;
	x2: number;
	y2: number;
	static newInterface(no, name, description,from,to,phenomenonList,x1,y1,x2,y2){
		let int = new Interface()
		int.no=no
		int.name=name
		int.description=description
		int.from=from
		int.to=to
		if(phenomenonList)
		int.phenomenonList=phenomenonList
		else 
		int.phenomenonList = []
		int.x1=x1
		int.y1=y1
		int.x2=x2
		int.y2=y2
		return int
	}
	static newInterfaceWithOld(old, description){
		let int = new Interface()
		int.no=old.no
		int.name=old.name
		int.description=description
		int.from=old.from
		int.to=old.to
		if(old.phenomenonList)
		int.phenomenonList=old.phenomenonList
		else 
		int.phenomenonList = []
		int.x1=old.x1
		int.y1=old.y1
		int.x2=old.x2
		int.y2=old.y2
		return int
	}
	static copyInterface(old){
		let int = new Interface()
		int.no=old.no
		int.name=old.name
		int.description=old.description
		int.from=old.from
		int.to=old.to
		if(old.phenomenonList)
		int.phenomenonList=old.phenomenonList
		else 
		int.phenomenonList = []
		int.x1=old.x1
		int.y1=old.y1
		int.x2=old.x2
		int.y2=old.y2
		return int
	}
	getNo(){return this.no}	
	setNo(no){this.no=no}

	getName(){return this.name}	
	setName(name){this.name=name}

	getDescription(){ return this.description}
	setDescription(description){ this.description=description}

	getFrom(){return this.from}
	setFrom(from){this.from=from}

	getTo(){return this.to}
	setTo(to){ this.to=to}

	getX1(){return this.x1}
	setX1(x1){ this.x1=x1}

	getY1(){return this.y1}
	setY1(y1){ this.y1=y1}

	getX2(){return this.x2}
	setX2(x2){ this.x2=x2}

	getY2(){return this.y2}
	setY2(y2){ this.y2=y2}
	getPhenomenonList(){return this.phenomenonList}
	setPhenomenonList(phenomenonList){ this.phenomenonList=phenomenonList}
	clearPhenomenonList(){	this.phenomenonList.length = 0	}
}